uamqp/error.py

```diff
@@ -248,3 +248,25 @@ def __init__(self, status_code, status_description):
         if self.description:
             message += u"\nDescription: {}".format(self.status_description.decode('utf-8'))
         super(TokenAuthFailure, self).__init__(message)
+
+
+class ErrorResponse(object):
+    """
+    """
+    def __init__(self, error_info=None, condition=None, description=None, info=None):
+        info = None
+        self.condition = condition
+        self.description = description
+        self.info = info
+        self.error = error_info
+        if isinstance(error_info, list) and len(error_info) >= 1:
+            if isinstance(error_info[0], list) and len(error_info[0]) >= 1:
```

Comment:
Is error_info completely arbitrary?
i.e. is there actual cases where error_info isn't a list? Or that error_info[0] isn't a list?

Reply:
I don't see the AMQP spec has mentioned special about the structure of "state" within a disposition frame -- in the spec, the "state" is of type "*" (Any).

So my understanding is that this is Azure specific, and these errors are conforming to the standard AMQP error structure (condition/description/info).